### PR TITLE
add the actual endpoints to get vessel details

### PR DESCRIPTION
**Pull request by fertandil87**
https://globalfishingwatch.atlassian.net/browse/VH-53 (the branch name is wrong sorry)

### CODE REVIEW

**Review by rdgfuentes — CHANGES_REQUESTED** [review]
Comments below. It would be great to start adding some tests too

File: applications/vessel-history/src/features/profile/Profile.tsx [hunk 1]
```diff
@@ -1,33 +1,102 @@
...
+        return error
+      })
+  }
+  useEffect(() => {
+    if (vesselID && dataset) {
+      fetchGFWData(vesselID, dataset)
+    }
+    if (tmtID) {
+      fetchTMTData(tmtID)
+    }
```
[the remark this answers is not in the document]
Comment (fertandil87) [reply applications/vessel-history/src/features/profile/Profile.tsx:71]: yes, this was a future improvement, I just added the endpoints to continue working then on that

File: applications/vessel-history/src/features/profile/Profile.tsx [hunk 2]
```diff
@@ -1,33 +1,102 @@
...
+    const tmtData = await GFWAPI.fetch<any>(url)
+      .then((json: any) => {
+        setVessel(json)
+        return null
+      })
+      .catch((error) => {
+        return error
+      })
+  }
+  const fetchTMTData = async (id: string) => {
```
[the remark this answers is not in the document]
Comment (fertandil87) [reply applications/vessel-history/src/features/profile/Profile.tsx:53]: that is another ticket

File: applications/vessel-history/src/features/profile/Profile.tsx [hunk 3]
```diff
@@ -1,33 +1,102 @@
...
+          icon="arrow-left"
+          className={styles.backButton}
+          onClick={async () => {
+            return
+          }}
+        ></IconButton>
+        {vessel && (
+          <h1>
+            Vessel Name
+            <p>+4 previous names</p>
```
[the remark this answers is not in the document]
Comment (fertandil87) [reply applications/vessel-history/src/features/profile/Profile.tsx:90]: managing and rendering data for another PR

File: applications/vessel-history/src/features/profile/Profile.tsx [hunk 4]
```diff
@@ -1,33 +1,102 @@
...
+    {
+      id: 'activity',
+      title: 'ACTIVITY',
+      content: <div />,
+    },
+    {
+      id: 'map',
+      title: 'MAP',
+      content: <MapWrapper vesselID={vesselID} setLastPosition={setLastPosition} />,
+    },
```
[the remark this answers is not in the document]
Comment (fertandil87) [reply applications/vessel-history/src/features/profile/Profile.tsx:38]: we will have a PR for this tabs

File: applications/vessel-history/src/features/profile/components/Info.tsx [hunk 5]
```diff
@@ -16,22 +16,13 @@ import MapButtton from './MapButtton'
...
-  const [vessel, setVessel] = useState(null)
-  const vessels = useSelector(selectVessels)
-  useMemo(() => {
-    GFWAPI.fetch(`/vessels/${vesselID}`)
-      .then((json: any) => {
-        setVessel(json)
-      })
-      .catch((error) => console.log(error))
-  }, [vesselID])
+  const vessel = props.vessel
```
[the remark this answers is not in the document]
Comment (fertandil87) [reply applications/vessel-history/src/features/profile/components/Info.tsx:25]: not understand

File: applications/vessel-history/src/features/profile/Profile.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (rdgfuentes) [reply applications/vessel-history/src/features/profile/Profile.tsx:37]: I'm saying that the content of the tabs Activity and MAP should be just a message with the text "Coming Soon!"

[the remark this answers is not in the document]
Comment (fertandil87) [reply applications/vessel-history/src/features/profile/Profile.tsx:37]: ok

File: applications/vessel-history/src/features/profile/components/Info.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (rdgfuentes) [reply applications/vessel-history/src/features/profile/components/Info.tsx:12]: I'm saying that we are not displaying any vessel data coming from the API. All the vessel data that this component is displaying is hardcoded.

**Pull request closed without merging**